extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/PrometheusEmitter.java

```diff
@@ -286,13 +290,16 @@ private void cleanUpStaleMetrics()
     Map<String, DimensionsAndCollector> map = metrics.getRegisteredMetrics();
     for (Map.Entry<String, DimensionsAndCollector> entry : map.entrySet()) {
       DimensionsAndCollector metric = entry.getValue();
-      if (metric.isExpired()) {
-        log.debug(
-            "Metric [%s] has expired (last updated [%d] ms ago)",
-            entry.getKey(),
-            metric.getMillisSinceLastUpdate()
-        );
-        metric.getCollector().clear();
+      Set<List<String>> labelValuesCopy = new HashSet<>(metric.getLabelValuesToStopwatch().keySet());
+      for (List<String> labelValues : labelValuesCopy) {
+        if (metric.removeIfExpired(labelValues)) {
+          log.debug(
+              "Metric [%s] with labels [%s] has expired",
+              entry.getKey(),
+              labelValues
+          );
+          metric.getCollector().remove(labelValues.toArray(new String[0]));
```

Comment:
Should the corresponding entry from `labelValuesToStopwatch` also be removed?  Otherwise, `labelValuesToStopwatch` can grow unbounded right?